Services/Statistics/SimulationStatistics.cs

```diff
@@ -44,25 +43,26 @@ public async Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string
             if (string.IsNullOrEmpty(simulationId))
             {
                 this.log.Error("Simulation Id cannot be null or empty");
-                return null;
+                throw new InvalidInputException("Simulation Id cannot be null or empty");
             }
 
-            string sqlCondition = " CONTAINS(ROOT.id, @simulationId)";
-            SqlParameter[] sqlParameters = new[] { new SqlParameter { Name = "@simulationId", Value = simulationId } };
             SimulationStatisticsModel statistics = new SimulationStatisticsModel();
 
             try
             {
-                var simulationRecords = (await this.simulationStatisticsStorage.GetAsync(sqlCondition, sqlParameters))
+                var simulationRecords = (await this.simulationStatisticsStorage.GetAllAsync())
                     .Select(p => JsonConvert.DeserializeObject<SimulationStatisticsRecord>(p.Data))
                     .ToList();
                  
                 foreach (var record in simulationRecords)
                 {
-                    statistics.TotalMessagesSent += record.Statistics.TotalMessagesSent;
-                    statistics.FailedDeviceConnections += record.Statistics.FailedDeviceConnections;
-                    statistics.FailedDevicePropertiesUpdates += record.Statistics.FailedDevicePropertiesUpdates;
-                    statistics.FailedMessages += record.Statistics.FailedMessages;
+                    if (record.SimulationId == simulationId)
```

Comment:
nit: you can apply this filter in line 53

Reply:
sure, done